fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -264,6 +267,19 @@ private Map<IndexOptions, List<LuceneDocumentFromRecord.DocumentField>> getIndex
     @SuppressWarnings("PMD.CloseResource")
     private void deleteDocument(Tuple groupingKey, Tuple primaryKey) throws IOException {
         final IndexWriter oldWriter = directoryManager.getIndexWriter(groupingKey, indexAnalyzerSelector.provideIndexAnalyzer(""));
+        final DirectoryReader directoryReader = DirectoryReader.open(oldWriter);
+        @Nullable final LucenePrimaryKeySegmentIndex segmentIndex = directoryManager.getDirectory(groupingKey).getPrimaryKeySegmentIndex();
+        if (segmentIndex != null) {
+            final LucenePrimaryKeySegmentIndex.DocumentIndexEntry documentIndexEntry = segmentIndex.findDocument(directoryReader, primaryKey);
+            if (documentIndexEntry != null) {
+                state.context.ensureActive().clear(documentIndexEntry.entryKey); // TODO: Only if valid?
```

Comment:
In practice, I think so, yes. Specifically, I believe that the NRT reader meets the criteria of being up-to-date with the writer. And I do not think it is possible for the document id to have been merged away (because that should replace the index entry with one for the new segment).

If we didn't try to fall back to the slow query, but gave an error, for instance, then this comment wouldn't apply. But if we do, there is a question of whether the existing index entry should be considered invalidated by that slow query delete.